src/app/components/cell.jsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react"
+
+const regex= new RegExp("^[1-9]+$")
+
+export default function Cell( { name } ) {
+
+    const [number, setNumber] = useState("");
+
+    var condition = "";
+
+    const handleChange = (e) => {
+        e.preventDefault();
+
+        setNumber(e.target.value);
+
+    }
+
+    return (
+        <input type="text" pattern="^[1-9]+$" value={number} className="w-10 h-10 border border-1 border-slate-400 text-center " onChange={(e) => handleChange(e)}></input>
```

Comment:
pattern="^[1-9]+$" the form validation pattern allows for integers of one and more digits. I think more correct pattern would be single digit pattern="[1-9]" Anyhow as form validation is happenin on form submit I think we need different approach something which will to react onChange or onInput.

Reply:
The regex expression chatGPT's fault 😊 you can reject it and I can fix it, but it's not something that will break the code so if you approve it I will change it later